src/utils_internal.rs

```diff
@@ -1,9 +1,15 @@
-use crate::errors_internal::{BleConnectionError, Error};
+use crate::errors_internal::Error;
+#[cfg(feature = "bluetooth-le")]
```

Comment:
Could you use `if cfg!(....) ` instead to have a multi-line condition?
https://doc.rust-lang.org/reference/conditional-compilation.html#the-cfg-macro

Reply:
> Could you use `if cfg!(....) ` instead to have a multi-line condition? https://doc.rust-lang.org/reference/conditional-compilation.html#the-cfg-macro

I don't think that's actually possible for use declarations; we could move the use declarations inside of the function that required it, but we'd still run afoul of situations where the elided use was part of the function signature eg

```rust
  #[cfg(feature = "bluetooth-le")]
  async fn scan_peripherals(adapter: &Adapter) -> Result<Vec<Peripheral>, btleplug::Error> {
    use btleplug::platform::{Adapter, Manager, Peripheral};

      adapter
          .start_scan(ScanFilter {
              services: vec![MSH_SERVICE],
          })
          .await?;
      adapter.peripherals().await
  }
```

Reply:
Sorry for a late response.

It pollutes the code right now but I'd vote for merging as it solves a problem for you. I can have a look for a better solution once I have time in about a month.

Reply:
If you have a better way to do it please feel free -- I changed my tools to use my fork of this repo so we've got time to come up with an 'optimal solution'

Reply:
I've actually worked with `#[cfg(...)]` at my job for the last week and while useful, some of its ergonomics is not that good. It's hard to improve on what you've done.